### ISSUE TITLE
Documentation is unclear

### ISSUE THREAD

**Issue by szentpali-janos** [mdn/content#36670]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

### What specific section or headline is this issue about?

Examples of access control scenarios

### What information was incorrect, unhelpful, or incomplete?

CORS is about a client and at least two servers. Throughout the examples it is unclear which server participates in the example transaction. There is a single example where the Host header hints at the identity of the server taking part in the transaction. As is the documentation is a guesswork for the reader, so not really a documentation. It would help immensely if the transaction diagrams and exchange request/responses would clearly state which server requests from which server.

### What did you expect to see?

Servers clearly identified both in drawings and transaction payloads.

### Do you have any supporting links, references, or citations?

No

### Do you have anything more you want to share?

No

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/http/cors`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/http/cors/index.md
* Last commit: https://github.com/mdn/content/commit/f75b2c86ae4168e59416aed4c7121f222afc201d
* Document last modified: 2024-10-01T22:01:14.000Z

</details>

**Comment by Josh-Cena** [comment]
> CORS is about a client and at least two servers.

Your confusion seems to be based on this, but I fail to see how this claim is true. CORS only involves one client and one server, where the client is on a different origin as the server. All our existing diagrams in the section in question only involve one client and one server because there are no more parties at play.

**Comment by szentpali-janos** [comment]
Sorry, but if there is one server and one client, why is CORS needed at all? Cross origin between which one server?

**Comment by Josh-Cena** [comment]
Cross-origin between the client and the server. For example you call `fetch("some-origin")` in a page that's on another origin.

**Comment by szentpali-janos** [comment]
So 
> in a page that's on another origin
does not mean a different domain (server) (or scheme)?

**Comment by Josh-Cena** [comment]
The scheme and domain are both parts of the origin.

For example, suppose the client-side JS of https://developer.mozilla.org sends a request to https://github.com. That would be a CORS request and subject to the CORS constraints.

**Comment by szentpali-janos** [comment]
OK, so while physically developer.mozilla.org github.com can be the same host, that is invisible to clients and are technically two server/domains.
And in the docs it would be nice to be clear which of those (or the code served from which of those) accesses which other.

**Comment by Josh-Cena** [comment]
I'll defer the decision to @wbamberg, but I think the current example is clear enough. There's only one server responding to requests (and handling CORS), which is the destination of `fetch`. While technically "developer.mozilla.org" is also a server, it does little more than responding to `GET` requests by responding with HTML/JS assets. It only functions as a client.

**Comment by wbamberg** [comment]
Josh is right. There's only one server in these example scenarios.

> Sorry, but if there is one server and one client, why is CORS needed at all? Cross origin between which one server?

Well sort of. You might say, the bit with the first server took place before these scenarios.

So unless I am very confused, this is what happens in full:

1. The user types `https://foo.org` into their address bar.
2. The browser makes a request to `https://foo.org`, gets back the response, and renders it.
3. Now the origin of this document in the browser (i.e. the client) is `https://foo.org` ("origin" literally means "the place this came from")
4. In this document, the user (for instance) clicks a button that wants to make a request to `https://bar.org`. This is a cross-origin request.
5. The browser starts the CORS process.

We might say, all these examples omit the first two steps here: they start with a document served from `https://foo.org`. All the interaction after that point is with `https://bar.org`. From that point on really, the `https://foo.org` might have exploded, and the rest of the flow would not be affected.

This is I guess implicit in bits like this:

> For example, suppose web content at https://foo.example wishes to fetch JSON content from domain https://bar.other. 

I wonder if "web content served from https://foo.example" might be better than "web content at https://foo.example"? I wonder if annotating the sequence diagrams to include the document's origin might help? I wonder if talking a bit more about the initial exchange with the first server might help?

**Comment by wbamberg** [comment]
What do you think about something like this:

<img width="1056" alt="Screen Shot 2024-11-08 at 11 09 53 AM" src="https://github.com/user-attachments/assets/6d8ebba5-c481-4131-857a-f6cec777e0af">

I am a little concerned that this doesn't distinguish the difference between the annotations, that "foo.example" is the origin and "bar.other" is the URL. I tried that but it looks a bit noisy:

<img width="1067" alt="Screen Shot 2024-11-08 at 11 12 40 AM" src="https://github.com/user-attachments/assets/31e0e190-352e-4b4c-aa7d-83eb3acc86c0">

Interested in what people think. If we don't think this is an improvement I think we should just close this issue, or just change:

> For example, suppose web content at https://foo.example/ wishes to fetch JSON content from domain https://bar.other/.

to "web content served from https://foo.example", but I think a change like that is unlikely to help people much.

**Comment by szentpali-janos** [comment]
That is very good as far as I am concerned.
Only thing that I'd do differently is the domain (name) of the origin (foo.example) and resource (bar.other) in that I'd use the same domain (names) as the ones at the beginning of the article: domain-a.com and domain-b.com.
Maybe the Origin: https://foo.exampe on the "Client" box is not necessary as it is implied by the "Origin" header below the "GET" request?

**Comment by wbamberg** [comment]
> Only thing that I'd do differently is the domain (name) of the origin (foo.example) and resource (bar.other) in that I'd use the same domain (names) as the ones at the beginning of the article: domain-a.com and domain-b.com.

Yeah, I agree we should do this too. For me at least `foo.example` and `bar.other` are better as names though (more obviously different) so I would prefer to use them everywhere.

**Comment by szentpali-janos** [comment]
Yes, I agree, for someone learning about CORS, using `foo.example` and `bar.other` vs. `domain-a` and `domain-b` might be less distracting/confusing. As long as it is used consistently.

**Comment by himanshugarg** [comment]
Do the drawings have any source files to modify?

**Comment by wbamberg** [comment]
Yes, they are at https://github.com/mdn/shared-assets/tree/main/images/diagrams/http/cors, some are Mermaid and some SVG. I did get around to working out how to update the Mermaid diagrams but then got distracted by some other task :/.

**Comment by wbamberg** [comment]
For https://github.com/mdn/content/issues/36670#issuecomment-2465576349 I think I did something like this:

```
%%{init: { "sequence": { "wrap": true, "width":250, "noteAlign": "left", "messageAlign": "center", "mirrorActors":false }} }%%

sequenceDiagram
    participant Client as Client<br><br>Origin#58; https#58;//foo.example
    participant Server as Server<br><br>URL#58; https#58;//bar.other

    Client->>Server: GET /resources/public-data/ HTTP/1.1<br/>Origin: https://foo.example
    Server->>Client: HTTP/1.1 200 OK<br/>Access-Control-Allow-Origin: *
```

(in case you are interested in taking this issue and would like the hint :)

**Comment by wbamberg** [comment]
Oh, also note I removed the mirrored actors, which should be done consistently in this page, if we want to do it.

**Comment by wbamberg** [comment]
I've filed https://github.com/mdn/shared-assets/pull/65 to update the diagrams.

**Comment by Josh-Cena** [comment]
Considering this done
